go/libraries/doltcore/sqle/index_lookup.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqle
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/src-d/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/row"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/schema"
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+type IndexLookupKeyIterator interface {
+	// NextKey returns the next key if it exists, and io.EOF if it does not.
+	NextKey(ctx *sql.Context) (row.TaggedValues, error)
+}
+
+type doltIndexLookup struct {
+	idx     DoltIndex
+	keyIter IndexLookupKeyIterator
+}
+
+func (il *doltIndexLookup) Indexes() []string {
+	return []string{il.idx.ID()}
+}
+
+// No idea what this is used for, examples aren't useful. From stepping through the code I know that we get index values
```

Comment:
I know what this is for now :)

In the meantime you can delete the comment, as it's not helpful.